### ISSUE TITLE
[strict-boolean-expressions] Consider adding a check for boolean return values

### ISSUE THREAD

**Issue by garth** [typescript-eslint/typescript-eslint#4002]
- [ ] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/strict-boolean-expressions": [
      "error",
      { "allowString": false, "allowNumber": false, "allowNullableObject": false }
    ]
  }
}
```

```TS
someArray.filter(thing => thing.id) // id of 0 accidentally removes item from array
```

**Expected Result**

It would be nice to be forced to convert to something like as this rule currently enforces with `if` and similar.

```ts
someArray.filter(thing => thing.id !== null)
```

**Actual Result**

Currently the rule does not check boolean return values.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.33.0` |
| `@typescript-eslint/parser`        | `4.33.0` |
| `TypeScript`                       | `4.4.3` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `16.10.0` |

**Comment by bradzacher** [comment]
duplicate of #1038
